### PR TITLE
DAOS-5409 test: Run parallel daos_test -F in CI

### PR DESCRIPTION
**Pull request by saurabhtandan**
Modified orterun run command to use clients instead of
using the test_runner node as client.
Also, made 'daos_test -F' to run in parallel with 16
processes over 2 client nodes.

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

### CODE REVIEW

File: src/tests/ftest/util/daos_core_base.py [hunk 1]
```diff
@@ -55,6 +58,13 @@ def setUp(self):
         self.update_log_file_names(self.subtest_name)
 
         super(DaosCoreBase, self).setUp()
+        self.hostfile_clients_slots = None
+        # single client for all tests except for 'daos_test -F'
+        if not self.subtest_name == "DAOS File System tests":
```
Comment (shimizukko) [inline src/tests/ftest/util/daos_core_base.py:63]:
```suggestion
        if self.subtest_name != "DAOS File System tests":
```

File: src/tests/ftest/util/daos_core_base.py [hunk 2]
```diff
@@ -103,13 +113,18 @@ def run_subtest(self):
         dmg = self.get_dmg_command()
         dmg_config_file = dmg.yaml.filename
 
+        # for 'daos_test -F' increase the number of client processes to 16
+        if subtest == "F":
+            num_clients = 16
```
Comment (shimizukko) [inline src/tests/ftest/util/daos_core_base.py:118]: This looks like test configuration. Does it make sense to set this value in daos_core_test.yaml?

File: src/tests/ftest/daos_test/daos_core_test_dfs.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    - server-B
+    - server-C
+    - server-D
+  test_clients:
+    - client-E
+    - client-F
+
+timeout: 600
+pool:
+  #This will create 8G of SCM and 16G of NVMe size of pool.
```
Comment (shimizukko) [inline src/tests/ftest/daos_test/daos_core_test_dfs.yaml:15]: Does it create a pool with 16G NVMe?

Comment (saurabhtandan) [reply src/tests/ftest/daos_test/daos_core_test_dfs.yaml:15]: Yes it will. 'daos_test' automatically created nvme of double the size of scm specified.

Comment (shimizukko) [reply src/tests/ftest/daos_test/daos_core_test_dfs.yaml:15]: I see. Thanks

File: src/tests/ftest/daos_test/daos_core_test_dfs.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# change host names to your reserved nodes, the
+# required quantity is indicated by the placeholders
+hosts:
+  test_servers:
+    - server-A
+    - server-B
+    - server-C
+    - server-D
+  test_clients:
```
Comment (mchaarawi) [inline src/tests/ftest/daos_test/daos_core_test_dfs.yaml:9]: is this the max number of client nodes you can use (i.e. the more the better)?

File: src/tests/ftest/daos_test/daos_core_test.yaml [hunk 5]
```diff
@@ -89,9 +89,6 @@ daos_tests:
     test_K:
       daos_test: K
       test_name: DAOS KV tests
-    test_F:
```
Comment (mchaarawi) [inline src/tests/ftest/daos_test/daos_core_test.yaml:92]: i think it's fine to keep this test running in daos_core_test with a single client.

File: src/tests/ftest/daos_test/daos_core_test_dfs.yaml (hunk 4 above)
Comment (mchaarawi) [reply src/tests/ftest/daos_test/daos_core_test_dfs.yaml:9]: not at the expense of the number of servers. was just thinking, like is 6 total the max?

Comment (saurabhtandan) [reply src/tests/ftest/daos_test/daos_core_test_dfs.yaml:9]: On Large cluster we have 8 total nodes so I can make total clients as 4. But you mentioned above you are fine to run this with single client which is a bit confusing. So you want me to have it run with single client multiple processes or multiple clients multiple processes?
Also note, currently we are running daos_core_test on medium cluster which has a max nodes of 4 and we are using them all for servers.
As this particular use case need to be run with more clients I moved only this test to large cluster.
If it's fine to have single client multiple processes, I can keep this within daos_core_test.py and just make the number of processes to 16.
But if it needs to be multi client multi processes, we need large hw cluster and I can increase the clients to 4. You may let me know if you want me to increase the num of processes from 16 to any other number in that case though.

Comment (mchaarawi) [reply src/tests/ftest/daos_test/daos_core_test_dfs.yaml:9]: no sorry that was not my intention. we do want to run this on multiple client nodes. I was asking here if we can do more than 2.
For my comment above, i didn't mean as a replacement. i just meant it's fine to let this test run twice, once with a single rank like it is today (don't need that change aboce) and once with this mode in parallel. but it's probably just redundant.

Comment (saurabhtandan) [reply src/tests/ftest/daos_test/daos_core_test_dfs.yaml:9]: yeah we can do that. I will leave it in daos_core_test as is. And will add 2 more client nodes for daos_core_test_dfs, which will make total clients as 4. Any suggestion for total number of client processes?

Comment (mchaarawi) [reply src/tests/ftest/daos_test/daos_core_test_dfs.yaml:9]: let's do 32 to exercise the conditionals more

Comment (saurabhtandan) [reply src/tests/ftest/daos_test/daos_core_test_dfs.yaml:9]: Done

**Comment by mchaarawi** [comment]
hmm actually the test failed.

**Comment by mchaarawi** [comment]
that's weird, the test just timeout, but no errors or timeouts in the logs..

**Review by shimizukko — APPROVED** [review]
LGTM

**Review by mchaarawi — APPROVED** [review]

**Pull request merged**